mapper/thin_edge_json/src/json.rs

```diff
@@ -288,6 +281,61 @@ mod tests {
             .ymd(2021, 04, 08)
             .and_hms(0, 0, 0)
     }
+    #[test]
+    fn test_str_with_invalid_timestamp() {
+        let input = r#"{
+            "time" : "2013-06-2217:03:14.000658767+02:00"
+        }"#;
+        let expected_error = r#"Invalid ISO8601 timestamp (expected YYYY-MM-DDThh:mm:ss.sss.±hh:mm): "2013-06-2217:03:14.000658767+02:00": input contains invalid characters"#;
+        let output_err = ThinEdgeJson::from_str(input, test_timestamp()).unwrap_err();
+        assert_eq!(output_err.to_string(), expected_error);
+    }
+
+    #[test]
+    fn test_str_with_valid_timestamp() {
+        let input = r#"{
+            "time" : "2021-04-30T17:03:14+02:00",
+            "temperature" : 25
+        }"#;
+        let timestamp = FixedOffset::east(2 * 3600)
+            .ymd(2021, 04, 30)
+            .and_hms(17, 03, 14);
+
+        let output = ThinEdgeJson::from_str(input, timestamp).unwrap();
+        assert_eq!(output.timestamp, timestamp);
+    }
+    #[test]
+    fn test_str_with_millisecond_timestamp() {
+        let input = r#"{
+            "time" : "2021-04-30T17:03:14.123+02:00",
+            "temperature" : 25
+        }"#;
+        let timestamp = FixedOffset::east(2 * 3600)
+            .ymd(2021, 04, 30)
+            .and_hms_milli(17, 03, 14, 123);
+
+        let output = ThinEdgeJson::from_str(input, timestamp).unwrap();
+        assert_eq!(output.timestamp, timestamp);

```

Comment:
I agree that the default timestamp passed as an argument to the `from_str` method is a bit misleading, but by using the very same timestamp as a default and as the input, this test is not necessarily testing what we expect to test since we have no way to be sure that the outcome is what has been parsed. In the example I provided, I used *two* different timestamps for that reason.